### PR TITLE
Consolidate dependency management on npm

### PR DESCRIPTION
**Pull request by dlabrecq**
This PR consolidates the angular-patternfly dependency management on npm for the pf 4 branch.

https://patternfly.atlassian.net/browse/PTNFLY-2035

Note that I'm using lodash-amd in package.json. The npm version of lodash is written for Node.js and uses require to load.

Also note that lodash no longer supports _.findWhere or _.remove. I had to replace functionality in filter-directive.js with _.find and _.filter.

### CODE REVIEW

**Comment by dtaylor113** [comment]
Hi, LGTM.  Our README.md needs to be updated, it still refers to:

```
<!-- Angular -->
<script src="bower_components/angular/angular.min.js"></script>

<!-- Angular-Bootstrap -->
<script src="bower_components/angular-bootstrap/ui-bootstrap.min.js"></script>
<script src="bower_components/angular-bootstrap/ui-bootstrap-tpls.min.js"></script>
```

Thanks,
- Dave

**Comment by dlabrecq** [comment]
Thanks, Dave. I updated the readme as well.

**Comment by dtaylor113** [comment]
I think you missing the 'd' in 'dependency' for this PR title :smile:  thank

**Comment by dlabrecq** [comment]
Anyone else want to review? @bleathem @dgutride

**Comment by dlabrecq** [comment]
Last chance to review. Going once, twice...

File: .travis.yml [hunk 1]
```diff
@@ -10,11 +10,14 @@ env:
     - TRIGGER_REPO_BRANCH: "master"
 
 before_install:
-  - 'git checkout -B $TRAVIS_BRANCH' # Reconcile detached HEAD
-  - 'npm install -g bower grunt-cli'
+  - if [[ `npm -v` != 3* ]]; then npm i -g npm@3; fi
+  - npm install -g bower grunt-cli
```
Comment (waldenraines) [inline .travis.yml]: Do we still need to install bower here if we're no longer using it?

Comment (dlabrecq) [inline .travis.yml]: We don't need bower to build, but Travis still needs to verify the bower install during the release process. In order for the Travis build to succeed, we still need this line -- for now.

FYI, the Travis build (i.e., the scripts from patternfly-eng-release) will still perform an npm and bower install for all our repos. That is, if there is a bower.json file available. During a release, the scripts will also perform an npm and bower install to verify.

Comment (waldenraines) [inline .travis.yml]: 👍

**Pull request merged**